src/wazuh_db/wdb_delta_event.c

```diff
@@ -1,425 +1,283 @@
-                        if (!wdb_dbsync_stmt_bind_from_string(stmt, column->value.index, column->value.type, curr, treat_as_null)) {
-                            merror(DB_AGENT_SQL_ERROR, wdb->id, sqlite3_errmsg(wdb->db));
-                            has_error = true;
-                        }
-                        if (column->next) {
-                            curr = next;
-                            next = strchr(curr, *FIELD_SEPARATOR_DBSYNC);
-                        }
-                    }
-                }
-            }
-
-            ret_val = !has_error && SQLITE_DONE == wdb_step(stmt);
-            os_free(data_temp);
-        } else {
-            merror(DB_CACHE_NULL_STMT);
+STATIC cJSON * wdb_dbsync_get_field_default(const struct field * field) {
+
+    cJSON * retval = NULL;
+
+    if (NULL != field) {
+        switch (field->type) {
+        case FIELD_INTEGER:
+            retval = cJSON_CreateNumber(field->default_value.integer);
+            break;
+        case FIELD_TEXT:
+            retval = cJSON_CreateString(field->default_value.text);
+            break;
+        case FIELD_INTEGER_LONG:
+            retval = cJSON_CreateNumber(field->default_value.integer_long);
+            break;
+        case FIELD_REAL:
+            retval = cJSON_CreateNumber(field->default_value.real);
+            break;
+        default:
+            mdebug2("Invalid syscollector field type: %i", field->type);
+            break;
         }
     }
-    return ret_val;
+
+    return retval;
 }
 
-bool wdb_modify_dbsync(wdb_t * wdb, struct kv const *kv_value, const char *data)
-{
-    const char * treat_as_null[] = {"", NULL};
+bool wdb_upsert_dbsync(wdb_t * wdb, struct kv const * kv_value, cJSON * data) {
     bool ret_val = false;
     if (NULL != data && NULL != wdb && NULL != kv_value) {
-        char query[OS_SIZE_2048] = { 0 };
-        strcat(query, "UPDATE ");
-        strcat(query, kv_value->value);
-        strcat(query, " SET ");
-
-        const size_t separator_count = os_strcnt(data, *FIELD_SEPARATOR_DBSYNC);
-        const size_t field_values_size = sizeof(char *) * (separator_count > 0 ? separator_count : 1);
-        // field_values vector will be used to point to the beginning of each field.
-        char ** field_values = NULL;
-        os_calloc(1, field_values_size + sizeof(char *), field_values);
-        char **curr = field_values;
-
-        char * data_temp = NULL;
-        os_strdup(data, data_temp);
-        *curr = data_temp;
+        char query[QUERY_MAX_SIZE] = {0};
+        int query_actual_size = 0;
 
-        char *curr_data = data_temp;
-        char *next = strchr(curr_data, *FIELD_SEPARATOR_DBSYNC);
-        // This loop replace '|' with '\0' and assign the beggining of each string to field_values vector.
-        while (NULL != next) {
-            *curr = curr_data;
-            *next++ = '\0';
+        query_actual_size += snprintf(query + query_actual_size, QUERY_MAX_SIZE - query_actual_size - 1,
```

Comment:
Nitpick: we could generate a helper method for this because it's repeated many times, but it isn't a priority.